### PR TITLE
Design of the training master of DLRover

### LINKED ISSUE #46: Design of EasyDL master

### PR DESCRIPTION
**Pull request by workingloong**
Fix #46

### CODE REVIEW

File: docs/design/easydl-master.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
+# Training Master For EasyDL
+
+The design describes how the architecture of the training master of EasyDL.
+When the user submit an ElasticJob,
```
Comment (nash635) [inline docs/design/easydl-master.md:4]: submits

File: docs/design/easydl-master.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+the ElasticJob controller will first create a training master.
+The master provides the following services:
+- Dynamic data sharding. The master can dispatch training data shards
+to workers according to the computation capability of the worker.
+- Collect runtime statistics, including the workload of each Pod,
+training throughput, training process.
+- Scale up/down Pods. The master generate a Scale CRD to scale up/down
```
Comment (nash635) [inline docs/design/easydl-master.md:11]: generate

File: docs/design/easydl-master.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+distributed systems.
+
+## Architecture of the Training Master
+
+The training master contains 5 components:
+- Resource Optimizer: it generates resource configuration plans for the job.
+- Scaler: it generates Scale CRDs according to scaling plans.
+- Stats Collector: it collects the runtime statistics for the job, including
+the resource usage of each Node, training throughput and the global step.
+- Data Shard Manager: it split the trianing dataset into shards and dispatch
```
Comment (nash635) [inline docs/design/easydl-master.md:25]: trianing -> training

File: docs/design/easydl-master.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+shards to workers. Eash shard only contains indices of training samples not
+the data of samples.
+- Node Watcher: it watches events of nodes and notify Data Shard Manager to
```
Comment (nash635) [inline docs/design/easydl-master.md:28]: notify -> notifies

File: docs/design/easydl-master.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+recover the shard of failed workers or notify Scaler to add the memory
+of OOM nodes.
+
+The design of the tarining master is shown as
```
Comment (nash635) [inline docs/design/easydl-master.md:32]: trianing -> training

**Review by nash635 — APPROVED** [review]

File: docs/design/easydl-master.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
+# Training Master For DLRover
+The design describes how the architecture of the training master of DLRover.
+When the user submits an ElasticJob,
```
Comment (samplise) [inline docs/design/easydl-master.md:3]: It is better to briefly introduce the master's main role or function in dlrover at first. Especially, to specify its different roles with/without the brain.

Comment (workingloong) [reply docs/design/easydl-master.md]: Done

**Review by samplise — APPROVED** [review]

**Pull request merged**